src/thermo/Phase.cpp

```diff
@@ -256,42 +256,122 @@ std::string Phase::speciesSPName(int k) const
     return m_name + ":" + speciesName(k);
 }
 
+vector<std::string> Phase::defaultState() const
+{
+    if (isPure()) {
+        if (isCompressible()) {
+            return {"T", "density"};
+        } else {
+            return {"T", "P"};
+        }
+    } else {
+        if (isCompressible()) {
+            return {"T", "density", "Y"};
+        } else {
+            return {"T", "P", "Y"};
+        }
+    }
+}
+
+vector<std::string> Phase::fullStates() const
+{
+    if (isPure()) {
+        if (isCompressible()) {
+            return {"TD", "TP", "UV", "DP", "HP", "SP", "SV"};
+        } else {
+            return {"TP", "HP", "SP"};
+        }
+    } else {
+        if (isCompressible()) {
+            return {"TDX", "TDY", "TPX", "TPY", "UVX", "UVY", "DPX", "DPY",
+                    "HPX", "HPY", "SPX", "SPY", "SVX", "SVY"};
+        } else {
+            return {"TPX", "TPY", "HPX", "HPY", "SPX", "SPY"};
+        }
+    }
+}
+
+vector<std::string> Phase::partialStates() const
+{
+    if (isPure()) {
+        return {};
+    } else {
+        if (!isCompressible()) {
```

Comment:
This is the only one of these functions where you negate this check, I think it'd be better to be consistent with the other functions.

Reply:
Not sure that I am following. This function returns state setters that don't affect molecular concentrations (which does not apply to pure substances).

Reply:
ah. I guess I was overthinking here!